src/commonMain/kotlin/io/github/koalaplot/core/bar/CandleStickPlotEntry.kt

```diff
@@ -0,0 +1,218 @@
+
+import androidx.compose.animation.core.AnimationSpec
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.*
+import io.github.koalaplot.core.style.KoalaPlotTheme
+import io.github.koalaplot.core.util.ExperimentalKoalaPlotApi
+import io.github.koalaplot.core.xygraph.XYGraphScope
+
+/**
+ * Represents a set of data for a single candle in a [CandleStickPlot].
+ *
+ * @param X The type of the x-axis values
+ * @param Y The type of the y-axis values
+ */
+public interface CandleStickPlotEntry<X, Y> {
+    /**
+     * The x-axis value of this [CandleStickPlotEntry].
+     */
+    public val x: X
+
+    /**
+     * The opening price of the candle.
+     */
+    public val open: Y
+
+    /**
+     * The closing price of the candle.
+     */
+    public val close: Y
+
+    /**
+     * The highest price of the candle.
+     */
+    public val high: Y
+
+    /**
+     * The lowest price of the candle.
+     */
+    public val low: Y
+}
+
+/**
+ * Returns an instance of a [CandleStickPlotEntry] using the provided data.
+ */
+public fun <X, Y> candleStickPlotEntry(x: X, open: Y, close: Y, high: Y, low: Y): CandleStickPlotEntry<X, Y> =
+    object : CandleStickPlotEntry<X, Y> {
+        override val x = x
+        override val open = open
+        override val close = close
+        override val high = high
+        override val low = low
+    }
+
+public const val WICKWIDTH: Float = 0.1f
+
+
+@ExperimentalKoalaPlotApi
+@Composable
+public fun <X, Y : Comparable<Y>> XYGraphScope<X, Y>.CandleStickPlot(
+    defaultCandle: @Composable BarScope.(entry: CandleStickPlotEntry<X, Y>) -> Unit = { _ -> },
+    modifier: Modifier = Modifier,
+    candleWidth: Float = 0.5f,
+    wickWidth: Float = WICKWIDTH,
+    animationSpec: AnimationSpec<Float> = KoalaPlotTheme.animationSpec,
+    content: CandleStickPlotScope<X, Y>.() -> Unit
+) {
+    val scope = remember(content, defaultCandle) { CandleStickPlotScopeImpl<X, Y>(defaultCandle) }
+    val data = remember(scope) {
+        scope.content()
+        scope.data
+    }
+
+    val candleBodyEntries = data.map { entry ->
+        verticalBarPlotEntry(entry.x, entry.open, entry.close)
+    }
+
+    val candleWickEntries = data.map { entry ->
+        verticalBarPlotEntry(entry.x, entry.low, entry.high)
+    }
+
+     VerticalBarPlot(
+        data = candleBodyEntries,
+        modifier = modifier,
+        barWidth = candleWidth,
+        animationSpec = animationSpec,
+        bar = { index ->
+            val entry = data[index]
+            val candleColor = if (entry.close >= entry.open) Color.Green else Color.Red
```

Comment:
The bar Composables should be exposed to the caller so they can (potentially) customize each bar, and choose the colors to use.